### PR TITLE
WebGPURenderer: implement compileAsync()

### PR DESCRIPTION
**Pull request by aardgoose**
Add `compileAsync( scene, camera )` for WebGPU and WebGL backends.

With WebGL the `KHR_parallel_shader_compile` extension is used if available.

Added to webgpu_occlusion example to test/demonstrate use.

Note: targetScene support to be added

### CODE REVIEW

File: examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ class WebGPUPipelineUtils {
 		const depthStencilFormat = utils.getCurrentDepthStencilFormat( renderObject.context );
 		const sampleCount = utils.getSampleCount( renderObject.context );
 
-		pipelineData.pipeline = device.createRenderPipeline( {
+		const piplineDescriptor = {
```
Comment (RenaudRohlinger) [inline examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js:107]: TYPO: `piplineDescriptor` -> `pipelineDescriptor`

Comment (aardgoose) [reply examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js]: fixed, thanks!

**Comment by RenaudRohlinger** [comment]
Looks good to me.
Just added a commit to forward the `compile -> compileAsync` via getter.

In the `WebGLRenderer` we throttle the check of completion with `setTimeout(()=>{}, 10)`. In my opinion using a RAF is just fine so I would let it as it is in the WebGLBackend.

I will let you decide if we merge this one or not as it affects the WebGPU side @sunag.

**Comment by sunag** [comment]
I think that since `compileAsync()` will call `updateBefore()` in many situations we will have multiples `render()` calls being executed during a `compileAsync()` process. I think `compileAsync()` should force the compilation of the entire consecutive process of the render tree and avoid overloading the frame in which it was called that could happen in some cases.

**Comment by aardgoose** [comment]
@RenaudRohlinger I Overwrote you commit by mistake, To you want to re add.

**Comment by RenaudRohlinger** [comment]
@aardgoose if I remember correctly the only difference was that I replaced:
```js
async compile( /*scene, camera, targetScene = null */ ) {

		console.warn( 'THREE.Renderer: .compile() is not implemented yet.' );

}
```
to:
```js
get compile() { // @deprecated, r170

      console.warn( 'THREE.WebGPURenderer: The property .compile has been deprecated. Use .compileAsync instead.' );
      return this.compileAsync;

}
```
In order to follow the best practice of the repo.

Are we good regarding the comment of @sunag?

File: examples/jsm/renderers/webgl/WebGLBackend.js [hunk 2]
```diff
@@ -47,6 +47,7 @@ class WebGLBackend extends Backend {
 		this.discard = false;
 
 		this.extensions.get( 'EXT_color_buffer_float' );
+		this.parallel = this.extensions.get( 'KHR_parallel_shader_compile' );
```
Comment (LeviPesin) [inline examples/jsm/renderers/webgl/WebGLBackend.js:50]: I don't really like that we are introducing a new property just to get a constant... Maybe at least mark it private?

File: examples/jsm/renderers/common/Renderer.js [hunk 3]
```diff
@@ -1048,10 +1168,28 @@ class Renderer {
...
+		this._bindings.updateForRender( renderObject );
+
+		this._pipelines.getForRender( renderObject, this._compilationPromises );
+
+	}
 
 	get compute() {
 
-		return this.computeAsync;
+		console.warn( 'THREE.Renderer: compile() is deprecated and will be removed in r170, use compileAsync instead.' ); // @deprecated, r170
```
Comment (LeviPesin) [inline examples/jsm/renderers/common/Renderer.js:1191]: I see that all other non-async methods just point to their async counterparts, why should we deprecate this one? /ping @sunag  @aardgoose @RenaudRohlinger

**Pull request merged**